packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +570,655 @@ class _CupertinoAlertActionSection extends StatelessWidget {
   // don't have many actions.
   final ScrollController scrollController;
 
-  bool get _layoutActionsVertically => children.length > 2;
+  @override
+  _CupertinoAlertActionSectionState createState() {
+    return new _CupertinoAlertActionSectionState();
+  }
+}
+
+class _CupertinoAlertActionSectionState extends State<_CupertinoAlertActionSection> {
+  static _CupertinoAlertActionSectionState of(BuildContext context) {
+    return context.ancestorStateOfType(
+      const TypeMatcher<_CupertinoAlertActionSectionState>(),
+    );
+  }
+
+  final Set<int> _pressedButtons = new Set<int>();
+
+  void onButtonDown(int index) {
+    setState(() => _pressedButtons.add(index));
+  }
+
+  void onButtonUp(int index) {
+    setState(() => _pressedButtons.remove(index));
+  }
```

Comment:
this seems rather sketchy. What happens if while the finger is down, the buttons switch sides? won't it now think the other button is pressed?

Reply:
Switched this logic over to ParentData.